### PR TITLE
Feature/orcomp604

### PR DESCRIPTION
**Pull request by ceo1647**
keyboard focus will proceed to another editing element if no choices of text input left for current element...for example
if we write "2" in hours and format AM/PM we will proceed to another editor... because no options left...
or we writing 4 in a days...-> or 3 if we have format where Month before days and month is February

Also if user combine mouse and keyboard input -> we proceed to the next editor on each user choice from combobox
image

### CODE REVIEW

File: src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs [hunk 1]
```diff
@@ -388,6 +390,8 @@ public override void OnApplyTemplate()
                 throw Log.ErrorAndCreateException<InvalidOperationException>("Can't find template part 'PART_MinuteNumericTextBox'");
             }
             _minuteNumericTextBox.ValueChanged += OnMinuteValueChanged;
+            _minuteNumericTextBox.TextChanged += OnMinuteTextChanged;
+            _minuteNumericTextBox.TextInput += (sender, args) => { };
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs:394]: Why this code? Debug left-over?

File: src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs [hunk 2]
```diff
@@ -1675,6 +1778,30 @@ private CultureInfo GetCulture()
...
+                return;
+            }
+
+            var text = secondOrMinuteTextBox.Text;
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                return;
+            }
+
+            if (new HashSet<string> { "1", "2", "3", "4", "5" }.Contains(text))
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs:1795]: Please create static hashset, we don't have to re-initialize a new array / hashset every time.

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]
Please see comments. Bonus points if a few unit tests were added.

File: src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs (hunk 1 above)
Comment (ceo1647) [reply src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs:394]: yes...thanks

File: src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs (hunk 2 above)
Comment (ceo1647) [reply src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs:1795]: fixed

File: src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs (hunk 1 above)
Comment (ceo1647) [reply src/Orc.Controls/Controls/DateTimePicker/DateTimePicker.cs:394]: removed

**Pull request merged**